### PR TITLE
Async messages handling via Laravel's queue feature

### PR DESCRIPTION
**Pull request by lezhnev74**
This PR has new LaravelQueueMessageProducer which can be used with built-in AsyncSwitchMessageRouter. 
Config example:

```php
... //config/prooph.php
        'command_bus' => [
            'router' => [
                'routes' => [],
                'type' => DirectRouter::class, // any default router
                'async_switch' => LaravelQueueMessageProducer::class, // PR's message producer
            ],
        ],
//...
```

The producer will dispatch Laravel's JOB which will be handled through the built-in Laravel's queue subsystem.

## Tests
There is 1 test which makes sure, that async command is passed to CommandBus through the LaravelAsyncMessageProducer

Opinions, suggestions?

### CODE REVIEW

**Comment by lezhnev74** [comment]
Ok, it failed building with:
```
Queue/HandleMessageJob.php (method_separation, no_trailing_whitespace, no_whitespace_in_blank_line, ordered_imports)
```

Is it a problem with formating of the sources? Do you know if there is a plugin for PHPStorm which can automatically format the source text?

**Comment by bweston92** [comment]
Use "php-cs-fixer fix"

**Comment by lezhnev74** [comment]
I wonder if there is a way to run this cs-fix script on each `git commit` so I don't have to. Is that a case?

**Comment by sandrokeil** [comment]
@lezhnev74 You can use [local git hooks](https://www.atlassian.com/git/tutorials/git-hooks#local-hooks).

**Comment by codeliner** [comment]
LGTM, thx @lezhnev74 
@bweston92 can we merge?

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 /vendor
 composer.lock
 .php_cs.cache
+.idea/
```
Comment (bweston92) [inline .gitignore]: Just add this to your local global git ignore.

File: src/Queue/LaravelQueueMessageProducer.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+use Prooph\Common\Messaging\Message;
+use Prooph\ServiceBus\Async\MessageProducer;
+use Prooph\ServiceBus\Exception\RuntimeException;
+use React\Promise\Deferred;
+
+final class LaravelQueueMessageProducer implements MessageProducer
+{
+    public function __invoke(Message $message, Deferred $deferred = null): void
+    {
+
```
Comment (bweston92) [inline src/Queue/LaravelQueueMessageProducer.php]: Remove empty line.

**Comment by bweston92** [comment]
I will look through and test when I get home.

File: src/Queue/LaravelQueueMessageProducer.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        // As far as I know - Laravel won't let me to get a promise, so no deferred queries are allowed
+        if ($deferred) {
+            throw new RuntimeException(__CLASS__ . ' cannot handle query messages which require future responses.');
+        }
+
+        // Now dispatch the Laravel job to deal with the queue
+        dispatch(new HandleMessageJob($message));
```
Comment (bweston92) [inline src/Queue/LaravelQueueMessageProducer.php]: Please inject via constructor: https://github.com/laravel/framework/blob/5.4/src/Illuminate/Contracts/Bus/Dispatcher.php

File: src/Queue/LaravelQueueMessageProducer.php [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+use React\Promise\Deferred;
+
+final class LaravelQueueMessageProducer implements MessageProducer
+{
+    public function __invoke(Message $message, Deferred $deferred = null): void
+    {
+
+        // As far as I know - Laravel won't let me to get a promise, so no deferred queries are allowed
+        if ($deferred) {
+            throw new RuntimeException(__CLASS__ . ' cannot handle query messages which require future responses.');
```
Comment (bweston92) [inline src/Queue/LaravelQueueMessageProducer.php]: Write a test case for this to be thrown.

**Comment by lezhnev74** [comment]
@sandrokeil thanks for the tip!

**Comment by lezhnev74** [comment]
@bweston92 I reworked it with your comments in mind. Let me know if something is missing

**Comment by bweston92** [comment]
Thanks for your work :)

**Pull request merged**